### ISSUE TITLE
Error when running on Firefox 47

### ISSUE THREAD

**Issue by XhmikosR** [sitespeedio/coach#119]
```
C:\>ver

Microsoft Windows [Version 6.1.7601]

C:\>node -v && npm -v
v6.2.0
3.8.9

C:\>webcoach --version
webcoach 0.22.4

C:\>webcoach https://www.sitespeed.io
Error running advice:  BrowserError: ECONNREFUSED connect ECONNREFUSED 127.0.0.1:58403
    at BrowsertimeError (C:\Users\xmr\AppData\Roaming\npm\node_modules\webcoach\node_modules\browsertime\lib\support\errors.js:5:5)
    at BrowserError (C:\Users\xmr\AppData\Roaming\npm\node_modules\webcoach\node_modules\browsertime\lib\support\errors.js:13:5)
    at startBrowser.call.catch.tap.tap.catch.catch (C:\Users\xmr\AppData\Roaming\npm\node_modules\webcoach\node_modules\browsertime\lib\core\seleniumRunner.js:73:15)
    at tryCatcher (C:\Users\xmr\AppData\Roaming\npm\node_modules\webcoach\node_modules\bluebird\js\release\util.js:16:23)
    at Promise._settlePromiseFromHandler (C:\Users\xmr\AppData\Roaming\npm\node_modules\webcoach\node_modules\bluebird\js\release\promise.js:502:31)
    at Promise._settlePromise (C:\Users\xmr\AppData\Roaming\npm\node_modules\webcoach\node_modules\bluebird\js\release\promise.js:559:18)
    at Promise._settlePromise0 (C:\Users\xmr\AppData\Roaming\npm\node_modules\webcoach\node_modules\bluebird\js\release\promise.js:604:10)
    at Promise._settlePromises (C:\Users\xmr\AppData\Roaming\npm\node_modules\webcoach\node_modules\bluebird\js\release\promise.js:679:18)
    at Async._drainQueue (C:\Users\xmr\AppData\Roaming\npm\node_modules\webcoach\node_modules\bluebird\js\release\async.js:138:16)
    at Async._drainQueues (C:\Users\xmr\AppData\Roaming\npm\node_modules\webcoach\node_modules\bluebird\js\release\async.js:148:10)
    at Immediate.Async.drainQueues [as _onImmediate] (C:\Users\xmr\AppData\Roaming\npm\node_modules\webcoach\node_modules\bluebird\js\release\async.js:17:14)
    at tryOnImmediate (timers.js:543:15)
    at processImmediate [as _immediateCallback] (timers.js:523:5)
```

**Comment by tobli** [comment]
Hi!
Does it work when using Chrome (passing -b chrome)? Seems to be an issue talking to Firefox, what version of Firefox are you running?

**Comment by XhmikosR** [comment]
I'm using Firefox v47b9 64-bit.

Using `-b chrome` it works fine indeed.

**Comment by XhmikosR** [comment]
BTW, this sounds weird


```
├─────────────────────────┼─────────────────────────────────────────────────────
──┼──────────┤
│ optimalCssSize          │ https://cdn.mpc-hc.org/assets/css/pack.71b13363.css
  │ 90       │
│                         │ size is 14.5 kB and that is bigger than the limit of
  │          │
│                         │ 14.5 kB. Try to make the CSS files fit into 14.5 kb.
  │          │
├─────────────────────────┼─────────────────────────────────────────────────────
```

**Comment by soulgalore** [comment]
yep, that seems like a bug, can you please file a separate issue. thanks!

**Comment by XhmikosR** [comment]
So it seems from a quick view from my AppCeyor branch, that Firefox 46.0.0.5955 works fine. Firefox 47.0.0.5990 fails for me. Not sure if it has to do with the fact that my FF is 64-bit. AppVeyor is using the 32-bit version as far as I can tell.

**Comment by XhmikosR** [comment]
All right, I verified it. The problem happens with the new FF 47beta version. 64-bit doesn't matter since I installed 46.0.1 64-bit and works fine.

Also, I doubt it's OS related, rather version related. But I'd need someone else to confirm.

**Comment by XhmikosR** [comment]
So, it seems it's indeed an issue with Firefox 47. Travis CI fails now too.

**Comment by soulgalore** [comment]
You are right. Yesterday 47 was released final, and it needs the Geckodriver to run and there was a new release yesterday https://github.com/mozilla/geckodriver/releases but there's no binaries yet. I'm not sure if the old geckodriver will work (haven't tested). @tobli started yesterday for a project to automatically install the geckodriver (as we do with Chromedriver). Also I think we need a new version of Selenium [2.54](https://github.com/SeleniumHQ/selenium/blob/master/javascript/node/selenium-webdriver/CHANGES.md) for "Removed the mandatory use of Firefox Dev Edition, when using Marionette driver".

**Comment by soulgalore** [comment]
Hopefully this will be fixed by 47.0.1. For 48 when they will drop support for the Firefoxdriver (the driver used by default by Selenum) we need to support Geckodriver. See https://github.com/sitespeedio/sitespeed.io/issues/993

**Comment by soulgalore** [comment]
Fixed in 47.0.1
